client/controller/index.web.js

```diff
@@ -40,6 +42,36 @@ export const ProviderWrappedLayout = ( {
 	const state = store.getState();
 	const userLoggedIn = isUserLoggedIn( state );
 
+	const [ isExperimentLoaded, setIsExperimentLoaded ] = useState( null );
+
+	async function loadDomainExperiment() {
+		const experimentAssignment = await loadExperimentAssignment(
+			'calypso_sidebar_upsell_dedicated_upgrade_flow_202301'
+		);
+		const variationName = experimentAssignment?.variationName;
+		sessionStorage.setItem( 'calypso_sidebar_upsell_experiment', variationName );
+		setIsExperimentLoaded( true );
+	}
+
+	const shouldLoadDomainExperiment = useCallback( () => {
+		const domainAndPackage = currentQuery && 'true' === currentQuery.domainAndPlanPackage;
+		const experimentPages =
+			window.location.pathname.startsWith( '/domains/add' ) ||
+			window.location.pathname.startsWith( '/plans/yearly' );
+
+		return domainAndPackage && experimentPages;
+	}, [ currentQuery ] );
+
+	useEffect( () => {
+		if ( shouldLoadDomainExperiment() ) {
+			loadDomainExperiment();
+		}
+	}, [ currentRoute, currentQuery, shouldLoadDomainExperiment ] );
+
+	if ( shouldLoadDomainExperiment() && ! isExperimentLoaded ) {
+		return null;
+	}
+
```

Comment:
```suggestion
    const isDomainAndPlanPackage = currentQuery && 'true' === currentQuery.domainAndPlanPackage;
    const isExperimentPages =
			window.location.pathname.startsWith( '/domains/add' ) ||
			window.location.pathname.startsWith( '/plans/yearly' );
    const isEligibleForExperiment = isDomainAndPlanPackage && isExperimentPages;
    
	const [ isExperimentLoaded, experimentAssignment ] = useExperimentAssignment(
	    'calypso_sidebar_upsell_dedicated_upgrade_flow_202301',
	    { isEligible: isEligibleForExperiment }
	);
	
	if ( ! isExperimentLoaded ) {
		return null;
	}
```

Reply:
> The error currently isn't fatal because for logged-in users, [we're not using SSR](https://github.com/Automattic/wp-calypso/blob/update/implement-domains-page-for-experiment/client/controller/index.node.js#L15).

If this is still causing SSR issues (if the path can be accessed while logged out,) you should add `userLoggedIn` to the isEligible check.